Cleanup grab-bag

jhorv opened the pull request:
A big pile of cleanup that should have no functional impact.  All tests passing.  Cherry-pick at will.

jeremydmiller commented on src/Marten/Events/Projections/Aggregator.cs:
```diff
@@ -9,7 +9,7 @@ namespace Marten.Events.Projections
 {
     public class Aggregator<T> : IAggregator where T : class, new()
     {
-        public static readonly string ApplyMethod = "Apply";
```
I'd rather we don't do that. I'm purposely using `static readonly` instead of `const`

jeremydmiller commented on src/Marten/Events/Projections/IAggregation.cs:
```diff
@@ -1,6 +1,6 @@
 namespace Marten.Events.Projections
 {
-    public interface IAggregation<TAggregate, TEvent>
+    public interface IAggregation<in TAggregate, in TEvent>
```
Let's definitely not do this one. I don't even want to let this kind of thing crop in right now until we absolutely have to have it for some unforeseeable use case.

jeremydmiller commented on src/Marten/Events/Projections/ITransform.cs:
```diff
@@ -1,7 +1,7 @@
 {
     // SAMPLE: ITransform
-    public interface ITransform<TEvent, TView>
+    public interface ITransform<TEvent, out TView>
```
See above.

jeremydmiller commented on src/Marten/Linq/Parsing/MethodCallParser.cs:
```diff
@@ -10,7 +10,7 @@ public abstract class MethodCallParser<T> : IMethodCallParser
         private readonly MethodInfo _method;
 
-        public MethodCallParser(Expression<Action<T>> method)
+        protected MethodCallParser(Expression<Action<T>> method)
```
Ugh, please definitely don't do that one. Way annoying with R# when you start implementing subclasses.

jeremydmiller commented on src/Marten/Linq/Parsing/StringComparisonParser.cs:
```diff
@@ -11,7 +11,7 @@ public abstract class StringComparisonParser : IMethodCallParser
         private readonly MethodInfo[] _supportedMethods;
 
-        public StringComparisonParser(params MethodInfo[] supportedMethods)
+        protected StringComparisonParser(params MethodInfo[] supportedMethods)
```
See above.

jhorv commented:
Want me to resubmit without those that you commented on, or do you still need to go through some of them?

phillip-haydon commented:
@jhorv you can make the changes in your current branch and this PR will update. No need to create a new branch or anything like that.

jhorv commented:
@phillip-haydon :

1. I reverted those disagreeable changes
2. Squashed all my individual commits down into 1 to make it easier to rebase
3. Rebased on current master
4. Pushed all of that to current branch

jeremydmiller commented:
@jhorv Honestly, I think I'd want you to issue them as separate PR's. You can't do an omnibus pull request like this with a bunch of unrelated things. I've got to tell you before you put any more work into this that I'm leaning toward not pulling this in.

jhorv commented:
Okay, I appreciate you saying so sooner than later.  I didn't expect it to be pulled in completely, and heck, they're mostly one-liners and I wouldn't mind if someone copied/pasted them in.  This was just the least-effort way of getting it in front of you for a minute.
